platform/x11/os_x11.cpp

```diff
@@ -127,10 +127,14 @@ void OS_X11::initialize(const VideoMode& p_desired,int p_video_driver,int p_audi
 	int xrandr_minor = 0;
 	int event_base, error_base;
 	xrandr_ext_ok = XRRQueryExtension(x11_display,&event_base, &error_base);
-	xrandr_handle = dlopen("libXrandr.so", RTLD_LAZY);
-	err = dlerror();
+	const char* sonames[3] = {"libXrandr.so.2", "libXrandr.so.2.2.0", "libXrandr.so"};
```

Comment:
The actual soname of the library is `libXrandr.so.2`. The other two are symlinks, and the `libXrandr.so.2.2.0` is bound to change relatively often, while `libXrandr.so.2` would be more stable.

So I'd advise to only check `libXrandr.so.2` (and maybe `libXrandr.so` if we want to be future proof for when the major is bumped to 3, but we'd still get the same issue at that time for people without the dev library installed anyway, so it might not be worth it).